File: Go/include/apple.go
package include

type Apple struct {
	X, Y int
}

func NewApple(snake *Snake) *Apple {
	apple := &Apple{0, 0}
	Eat(apple, snake)
	return apple
}

func Eat(apple *Apple, snake *Snake) {
	for range 1000 {
		blocked := false

		x, y := GetRandInt(0, 19), GetRandInt(0, 19)

		for curr := snake.Head; curr != nil; curr = curr.Next {
			if curr.X == x && curr.Y == y {
				blocked = true
				break
			}
		}

		if !blocked {
			apple.X = x
			apple.Y = y
			break
		}
	}
}

File: CSharp/include/Apple.cs
namespace Ouroboros.Include;

public class Apple
{
    public int X { get; private set; }

    public int Y { get; private set; }

    public Apple(Snake snake)
    {
        Eat(snake);
    }

    public void Eat(Snake snake)
    {
        for (int i = 0; i < 1000; ++i)
        {
            var pos = RNGMachine.GetRandomCoordinates(19, 19);

            if (!snake.GetNodes().Any(node => (node.X, node.Y) == pos))
            {
                (X, Y) = pos;
                return;
            }
        }
    }
}
